WordPressAuthenticator/NUX/NUXButton.swift

```diff
@@ -1,54 +1,157 @@
+    ///
+    @IBInspectable var isPrimary: Bool = false {
+        didSet {
+            configureBackgrounds()
+            configureTitleColors()
+        }
+    }
+
+
+    // MARK: - LifeCycle Methods
+
+    open override func didMoveToWindow() {
+        super.didMoveToWindow()
+        configureAppearance()
+    }
+
+    open override func awakeFromNib() {
+        super.awakeFromNib()
+        configureAppearance()
+    }
+
+    /// Setup: shorter reference for default style
+    ///
+    private let style = WordPressAuthenticatorStyle.defaultStyle
+
+    /// Setup: Everything = [Insets, Backgrounds, titleColor(s), titleLabel]
+    ///
+    private func configureAppearance() {
+        configureInsets()
+        configureBackgrounds()
+        configureTitleColors()
+        configureTitleLabel()
+    }
+
+    /// Setup: FancyButton's Default Settings
+    ///
+    private func configureInsets() {
+        contentEdgeInsets = contentInsets
+    }
 
+    /// Setup: BackgroundImage
+    ///
+    private func configureBackgrounds() {
         let normalImage: UIImage
-        let highlightImage: UIImage
-        let titleColorNormal: UIColor
+        let highlightedImage: UIImage
+        let disabledImage = UIImage.renderBackgroundImage(fill: style.disabledBackgroundColor, border: style.disabledBorderColor, shadowOffset: Metrics.backgroundShadowOffset)
 
         if isPrimary {
-            normalImage = .beveledBlueButtonImage
-            highlightImage = .belevedBlueButtonDownImage
-
-            titleColorNormal = UIColor.white
+            normalImage = UIImage.renderBackgroundImage(fill: style.primaryNormalBackgroundColor, border: style.primaryNormalBorderColor, shadowOffset: Metrics.backgroundShadowOffset)
+            highlightedImage = UIImage.renderBackgroundImage(fill: style.primaryHighlightBackgroundColor, border: style.primaryHighlightBorderColor, shadowOffset: Metrics.backgroundShadowOffset)
         } else {
-            normalImage = .beveledSecondaryButtonImage
-            highlightImage = .beveledSecondaryButtonDownImage
-
-            titleColorNormal = WPStyleGuide.darkGrey()
+            normalImage = UIImage.renderBackgroundImage(fill: style.secondaryNormalBackgroundColor, border: style.secondaryNormalBorderColor, shadowOffset: Metrics.backgroundShadowOffset)
+            highlightedImage = UIImage.renderBackgroundImage(fill: style.secondaryHighlightBackgroundColor, border: style.secondaryHighlightBorderColor, shadowOffset: Metrics.backgroundShadowOffset)
         }
 
-        let disabledImage = UIImage.beveledDisabledButtonImage
-        let titleColorDisabled = WPStyleGuide.greyLighten30()
-
         setBackgroundImage(normalImage, for: .normal)
-        setBackgroundImage(highlightImage, for: .highlighted)
+        setBackgroundImage(highlightedImage, for: .highlighted)
         setBackgroundImage(disabledImage, for: .disabled)
 
+        activityIndicator.activityIndicatorViewStyle = .gray
```

Comment:
Perhaps this should go into `configureActivityIndicator()` (since it's not really a Background?)

Reply:
✔️